fv3config/config/derive.py

```diff
@@ -42,33 +42,115 @@ def get_run_duration(config):
     the remote initial_conditions path in the given configuration dictionary.
 
     Args:
         config (dict): a configuration dictionary
+        coupler_res_date_getter (func): a function that reads either the
+            initialization_date or the current date from a coupler.res file
 
     Returns:
-        list: current_date as list of ints [year, month, day, hour, min, sec]
+        list: date as list of ints [year, month, day, hour, min, sec]
     """
     force_date_from_namelist = config["namelist"]["coupler_nml"].get(
         "force_date_from_namelist", False
     )
     # following code replicates the logic that the fv3gfs model uses to determine the current_date
     if force_date_from_namelist:
-        current_date = config["namelist"]["coupler_nml"].get(
-            "current_date", [0, 0, 0, 0, 0, 0]
-        )
+        date = config["namelist"]["coupler_nml"].get("current_date", [0, 0, 0, 0, 0, 0])
     else:
         coupler_res_filename = _get_coupler_res_filename(config)
         if coupler_res_filename is not None:
-            current_date = _get_current_date_from_coupler_res(coupler_res_filename)
+            date = coupler_res_date_getter(coupler_res_filename)
         else:
-            current_date = config["namelist"]["coupler_nml"].get(
+            date = config["namelist"]["coupler_nml"].get(
                 "current_date", [0, 0, 0, 0, 0, 0]
             )
-    return current_date
+    return date
+
+
+def get_current_date(config):
+    """Return current_date from configuration dictionary. This function may read from
+    the remote initial_conditions path in the given configuration dictionary.
+
+    Args:
+        config (dict): a configuration dictionary
+
+    Returns:
+        list: current_date as list of ints [year, month, day, hour, min, sec]
+    """
+    return _get_date(config, _get_current_date_from_coupler_res)
+
+
+def get_diag_table_base_date(config):
+    """Return base_date from configuration dictionary. This function may read from
+    the remote initial_conditions path in the given configuration dictionary.
+
+    In the event that force_date_from_namelist is specified in the config, this
+    function returns the current_date or a null date if that does not exist.  If the
+    date is not set to be forced from the namelist, this function first looks for the
+    existence of a coupler.res file, and if it exists, returns the initialization_date
+    written within it. If a coupler.res files does not exist, the current_date
+    specified in the namelist or a null date is used.  To FV3GFS, a null date and
+    the current date are equivalent.
+
+    Returning the initialization_date instead of the current_date if a coupler.res
+    file exists is important for obtaining reproducible restarts in segmented runs.
+    Another advantage is that it harmonizes the units used to encode times in
+    fotran-generated diagnostics files across segments. See more discussion in
+    GH 147.
```

Comment:
I find this docstring pretty hard to follow would be easier to just see the if-statements in the code.